### PR TITLE
Download infra-env events log

### PR DESCRIPTION
**Pull request by lalon4**
For better debugging, adding infraenv events collection.

### CODE REVIEW

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 1]
```diff
@@ -164,6 +164,12 @@ def download_logs(
             client.download_cluster_events(cluster["id"], get_cluster_events_path(cluster, output_folder))
             shutil.copy2(os.path.join(os.path.dirname(os.path.realpath(__file__)), "events.html"), output_folder)
 
+        infra_env_list = client.get_infra_env_by_cluster_id(cluster["id"])
```
Comment (eliorerz) [inline src/assisted_test_infra/download_logs/download_logs.py:167]: It's not better to iterate over the hosts infra_envs?

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 2]
```diff
@@ -279,7 +285,11 @@ def _must_gather_kube_api(cluster_name, cluster_deployment, agent_cluster_instal
 
 
 def get_cluster_events_path(cluster, output_folder):
-    return os.path.join(output_folder, f"cluster_{cluster['id']}_events.json")
+    return os.path.join(output_folder, f"cluster_{cluster['id']}_cluster_events.json")
```
Comment (eliorerz) [inline src/assisted_test_infra/download_logs/download_logs.py:288]: Need to change that. It already has a `cluster` prefix

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 3]
```diff
@@ -279,7 +285,11 @@ def _must_gather_kube_api(cluster_name, cluster_deployment, agent_cluster_instal
...
+
+
+def get_infraenv_events_path(infra_env_id, output_folder):
+    return os.path.join(output_folder, f"{infra_env_id}_infraenv_events.json")
```
Comment (eliorerz) [inline src/assisted_test_infra/download_logs/download_logs.py:292]:
```suggestion
    return os.path.join(output_folder, f"infraenv_{infra_env_id}_events.json")
```

File: src/service_client/assisted_service_api.py [hunk 4]
```diff
@@ -420,6 +420,15 @@ def download_cluster_events(self, cluster_id: str, output_file: str, categories=
             _file.write(json.dumps(events, indent=4).encode())
             self._events_junit_exporter.collect(events, suite_name=f"cluster_events_{cluster_id}")
 
+    def download_infraenv_events(self, infra_env_id: str, output_file: str, categories=None) -> None:
```
Comment (eliorerz) [inline src/service_client/assisted_service_api.py:423]: Missing `categories` type hint annotations

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 1 above)
Comment (lalon4) [reply src/assisted_test_infra/download_logs/download_logs.py:167]: I am getting all infra envs related to certain cluster - so i think it covers them all, no?

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 2 above)
Comment (lalon4) [reply src/assisted_test_infra/download_logs/download_logs.py]: done

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 3 above)
Comment (lalon4) [reply src/assisted_test_infra/download_logs/download_logs.py]: Done

File: src/service_client/assisted_service_api.py (hunk 4 above)
Comment (lalon4) [reply src/service_client/assisted_service_api.py]: Done

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 1 above)
Comment (lalon4) [reply src/assisted_test_infra/download_logs/download_logs.py]: done

**Review by eliorerz — APPROVED** [review]
LGTM

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**